json/access/src/main/java/ch/qos/logback/contrib/json/access/JsonLayout.java

```diff
@@ -0,0 +1,396 @@
+    protected boolean includeProtocol;
+    protected boolean includeContentLength;
+    protected boolean includeRequestURL;
+    protected boolean includeRemoteHost;
+    protected boolean includeServerName;
+    protected boolean includeRequestHeader;
+    protected boolean includeRequestParameter;
+    protected boolean includeLocalPort;
+    protected boolean includeRequestContent;
+    protected boolean includeResponseContent;
+
+
+    public JsonLayout() {
+        super();
+        this.includeRemoteAddr = true;
+        this.includeRemoteUser = true;
+        this.includeRequestTime = true;
+        this.includeRequestURI = true;
+        this.includeStatusCode = true;
+        this.includeMethod = true;
+        this.includeProtocol = true;
+        this.includeRequestURL = false;
+        this.includeContentLength = false;
+        this.includeRemoteHost = true;
+        this.includeServerName = true;
+        this.includeRequestHeader = true;
+        this.includeRequestParameter = true;
+        this.includeLocalPort = false;
+        this.includeRequestContent = false;
+        this.includeResponseContent = false;
+    }
+
+    @Override
+    protected Map toJsonMap(IAccessEvent event) {
+        Map<String, Object> map = new LinkedHashMap<String, Object>();
+
+        addTimestamp(TIMESTAMP_ATTR_NAME, this.includeTimestamp, event.getTimeStamp(), map);
+        add(REMOTEADDR_ATTR_NAME, this.includeRemoteAddr, event.getRemoteAddr(), map);
+        add(REMOTEUSER_ATTR_NAME, this.includeRemoteUser, event.getRemoteUser(), map);
+        addRequestTime(event.getElapsedTime(), map);
+        addInt(STATUSCODE_ATTR_NAME, this.includeStatusCode, event.getStatusCode(), map);
+        add(METHOD_ATTR_NAME, this.includeMethod, event.getMethod(), map);
+        add(REQUESTURI_ATTR_NAME, this.includeRequestURI, event.getRequestURI(), map);
+        add(PROTOCOL_ATTR_NAME, this.includeProtocol, event.getProtocol(), map);
+        add(CONTENTLENGTH_ATTR_NAME, this.includeContentLength, Long.toString(event.getContentLength()), map);
+        add(REQUESTURL_ATTR_NAME, this.includeRequestURL, event.getRequestURL(), map);
+        add(REMOTEHOST_ATTR_NAME, this.includeRemoteHost, event.getRemoteHost(), map);
+        add(SERVERNAME_ATTR_NAME, this.includeServerName, event.getServerName(), map);
+        addMap(REQUESTHEADER_ATTR_NAME, this.includeRequestHeader, event.getRequestHeaderMap(), map);
+        addMap(REQUESTPARAMETER_ATTR_NAME, this.includeRequestParameter, event.getRequestParameterMap(), map);
+        addInt(LOCALPORT_ATTR_NAME, this.includeLocalPort, event.getLocalPort(), map);
+        add(REQUESTCONTENT_ATTR_NAME, this.includeRequestContent,  event.getRequestContent(), map);
+        add(RESPONSECONTENT_ATTR_NAME, this.includeResponseContent,  event.getResponseContent(), map);
+
+        return map;
+    }
+
+    void addMap(String key, boolean field, Map<String, ?> mapValue, Map<String, Object> map) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/tony19/logback-contrib/file/2377134885/issues/source?bid=3222643&fileBranchId=3222645#l224)